pynxtools/dataconverter/readers/xps/reader.py

```diff
@@ -346,10 +367,24 @@ def read(self,
                             REPLACE_NESTED
                         )
                     )
-            elif file_ext in [".sle", ".xml"]:
+            elif file_ext in [".sle", ".xml", ".txt"]:
                 data_dict = XpsDataFileParser([file]).get_dict(**kwargs)
+
+                # If there are multiple input data files, make sure
+                # that existing keys are not overwritten.
+                existing = [
+                    (key, xps_data_dict[key], data_dict[key]) for
+                    key in data_dict if key in xps_data_dict
+                ]
+
                 xps_data_dict = {**xps_data_dict, **data_dict}
-                config_file = reader_dir.joinpath(f"config_file_{file_ext.rsplit('.')[1]}.json")
+                for (key, value1, value2) in existing:
```

Comment:
Here, the dictionaries are merged and the values for keys that are in both dictionaries are concatenated.